### PR TITLE
MULE-14532: fixed thread jump

### PR DESCRIPTION
**Pull request by mbuchwald**

### CODE REVIEW

File: core-tests/src/test/java/org/mule/runtime/core/internal/processor/strategy/ProactorStreamEmitterProcessingStrategyTestCase.java [hunk 1]
```diff
@@ -315,7 +315,6 @@ public void singleCpuLightConcurrentMaxConcurrency1() throws Exception {
                                                                                                     1, true)),
                        true, CPU_LITE, 1);
     assertThat(threads, hasSize(1));
-    assertThat(threads, not(hasItem(startsWith(CPU_LIGHT))));
```
Comment (elrodro83) [inline core-tests/src/test/java/org/mule/runtime/core/internal/processor/strategy/ProactorStreamEmitterProcessingStrategyTestCase.java]: if this is expected now, just remove the `not`

Comment (elrodro83) [reply core-tests/src/test/java/org/mule/runtime/core/internal/processor/strategy/ProactorStreamEmitterProcessingStrategyTestCase.java]: or better yet, count the occurrences as `ProactorStreamWorkQueueProcessingStrategyTestCase` does with the ring buffer threads

File: core/src/main/java/org/mule/runtime/core/internal/processor/strategy/ProactorStreamEmitterProcessingStrategyFactory.java [hunk 2]
```diff
@@ -133,12 +133,12 @@ public ProactorStreamEmitterProcessingStrategy(Supplier<Scheduler> ringBufferSch
     public Sink createSink(FlowConstruct flowConstruct, ReactiveProcessor function) {
       final long shutdownTimeout = flowConstruct.getMuleContext().getConfiguration().getShutdownTimeout();
       List<ReactorSink<CoreEvent>> sinks = new ArrayList<>();
-
       int concurrency = maxConcurrency < subscribers ? maxConcurrency : subscribers;
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/processor/strategy/ProactorStreamEmitterProcessingStrategyFactory.java]: doesn't this give the same result as `getParallelism()`?

Comment (mbuchwald) [reply core/src/main/java/org/mule/runtime/core/internal/processor/strategy/ProactorStreamEmitterProcessingStrategyFactory.java]: No. Subscribers, maxConcurrency and Parallelism are 3 different values. In the factory, the value given for parallelism is max(CORES, maxConcurrency), so it's at least as maxConcurrency. 
After trying different configurations this was the one that got best results in perf tests (and the one that takes into account maxConcurrency... if not, even if the user sets maxConcurrency = 1, the value won't be taken into account)

**Pull request merged**